x/chainmanager/types/params.go

```diff
@@ -89,45 +91,50 @@ func (p Params) String() string {
 
 // Validate checks that the parameters have valid values.
 func (p Params) Validate() error {
-	if err := validateHeimdallAddress("matic_token_address", p.ChainParams.MaticTokenAddress); err != nil {
+	if err := validateAccAddress(MaticTokenAddress, p.ChainParams.MaticTokenAddress); err != nil {
 		return err
 	}
 
-	if err := validateHeimdallAddress("staking_manager_address", p.ChainParams.StakingManagerAddress); err != nil {
+	if err := validateAccAddress(StakingManagerAddress, p.ChainParams.StakingManagerAddress); err != nil {
 		return err
 	}
 
-	if err := validateHeimdallAddress("slash_manager_address", p.ChainParams.SlashManagerAddress); err != nil {
+	if err := validateAccAddress(SlashManagerAddress, p.ChainParams.SlashManagerAddress); err != nil {
 		return err
 	}
 
-	if err := validateHeimdallAddress("root_chain_address", p.ChainParams.RootChainAddress); err != nil {
+	if err := validateAccAddress(RootChainAddress, p.ChainParams.RootChainAddress); err != nil {
 		return err
 	}
 
-	if err := validateHeimdallAddress("staking_info_address", p.ChainParams.StakingInfoAddress); err != nil {
+	if err := validateAccAddress(StakingInfoAddress, p.ChainParams.StakingInfoAddress); err != nil {
 		return err
 	}
 
-	if err := validateHeimdallAddress("state_sender_address", p.ChainParams.StateSenderAddress); err != nil {
+	if err := validateAccAddress(StateSenderAddress, p.ChainParams.StateSenderAddress); err != nil {
 		return err
 	}
 
-	if err := validateHeimdallAddress("state_receiver_address", p.ChainParams.StateReceiverAddress); err != nil {
+	if err := validateAccAddress(StateReceiverAddress, p.ChainParams.StateReceiverAddress); err != nil {
 		return err
 	}
 
-	if err := validateHeimdallAddress("validator_set_address", p.ChainParams.ValidatorSetAddress); err != nil {
+	if err := validateAccAddress(ValidatorSetAddress, p.ChainParams.ValidatorSetAddress); err != nil {
 		return err
 	}
 
 	return nil
 }
 
-func validateHeimdallAddress(key string, value sdk.AccAddress) error {
+func validateAccAddress(key string, value sdk.AccAddress) error {
 	if value.String() == "" {
 		return fmt.Errorf("Invalid value %s in chain_params", key)
 	}
+	//convert string to AccAddress and compare
+	addr := sdk.AccAddress([]byte(key))
+	if !addr.Equals(value) {
+		return fmt.Errorf("Key: %s is not equal to Value: %s", key, value)
+	}
```

Comment:
Why it should be equal to the key? This is not correct. We should validate based on the key, like if key is `SlashManagerAddress`, we should use some logic to validate it. Remove this and add a TODO to address this if it is hard to decide on the validation part.

Reply:
Got it. But I am not sure what logic to use. Hence I am adding TODO comment.